planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/utils.cpp

```diff
@@ -1247,4 +1252,57 @@ bool filter_target_lane_objects(
 
   return false;
 }
+
+std::vector<lanelet::ConstLanelets> get_preceding_lanes(const CommonDataPtr & common_data_ptr)
+{
+  const auto & route_handler_ptr = common_data_ptr->route_handler_ptr;
+  const auto & target_lanes = common_data_ptr->lanes_ptr->target;
+  const auto & ego_pose = common_data_ptr->get_ego_pose();
+  const auto backward_lane_length = common_data_ptr->lc_param_ptr->backward_lane_length;
+
+  const auto preceding_lanes_list =
+    utils::getPrecedingLanelets(*route_handler_ptr, target_lanes, ego_pose, backward_lane_length);
+
+  const auto & current_lanes = common_data_ptr->lanes_ptr->current;
+  std::unordered_set<lanelet::Id> current_lanes_id;
+  for (const auto & lane : current_lanes) {
+    current_lanes_id.insert(lane.id());
+
+    const auto prev_lanes = route_handler_ptr->getPreviousLanelets(lane);
+    for (const auto & prev_lane : prev_lanes) {
+      current_lanes_id.insert(prev_lane.id());
+    }
+  }
```

Comment:
Won't this logic unnecessarily insert duplicated lane ids for each lane in `current_lanes`

Also I can't see why you need to `getPreviousLanelets` for each lane in `current_lanes`, maybe it is sufficient to do it only for first lane?

Reply:
No this logic is necessary. Because there might be lanes that is connected to the current lanes, but not belong to the route.

![image](https://github.com/user-attachments/assets/42557072-a3c8-4e09-94a8-cdf2eae920b2)

For example these lanes is connected to current lanes.